### ISSUE TITLE
False positive for explicit_outlives_requirements lint

### ISSUE THREAD

**Issue by dbeckwith** [rust-lang/rust#60993]
```rust
#![allow(dead_code)]
#![deny(explicit_outlives_requirements)]

struct Test<'a, T>(&'a T) where T: ?Sized;

```

([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=26ea04280c65fc3bdb943a18af422b88))

Errors:

```
   Compiling playground v0.0.1 (/playground)
error: outlives requirements can be inferred
 --> src/lib.rs:4:19
  |
4 | struct Test<'a, T>(&'a T) where T: ?Sized;
  |                   ^^^^^^^^^^^^^^^^^^^^^^^ help: remove these bounds
  |
note: lint level defined here
 --> src/lib.rs:2:9
  |
2 | #![deny(explicit_outlives_requirements)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

error: Could not compile `playground`.

To learn more, run the command again with --verbose.

```

Maybe I'm missing something, but this lint error makes no sense in this context, as there are no outlives bounds written.

Curiously, `struct Test<'a, T: ?Sized>(&'a T);` compiles without error.
